src/status_im/communities/core.cljs

```diff
@@ -36,9 +36,9 @@
                     :chatId      :chat-id}))
 
 (defn <-requests-to-join-community-rpc
```

Comment:
This and the other singular version of this function would be better placed in the `data_store/*` namespaces. Something for a future PR if you want :)

Also this `<-requests-to-join-community-rpc` is doing 2 things that can be better described with specialized functions. The function is indexing by the result of `key-fn` and the value in the map is transformed by `<-request-to-join-community-rpc`.

For step 1 we should use `utils.collection/index-by`, and for step 2 we have Clojure's core function `update-vals`. Look at the implementation, it's basically what's written here, **but it performs better** because it uses transients. https://github.com/clojure/clojure/blob/2b3ba822815981e7f76907a3b75e9ecc428f7656/src/clj/clojure/core.clj#L8008

This is one of the reasons a love Clojure for data processing!

```clojure
(let [coll [{:id 1 :name "A"}
            {:id 2 :name "B"}
            {:id 3 :name "C"}]]
  (update-vals (index-by :id coll)
               #(assoc % :age (* (:id %) 10))))
;; => {1 {:id 1, :name "A", :age 10},
;;     2 {:id 2, :name "B", :age 20},
;;     3 {:id 3, :name "C", :age 30}}
```

Reply:
🤤

Reply:
Quick note: @J-Son89 found that `updated-vals` is not available in our CLJS version https://github.com/status-im/status-mobile/blob/develop/nix/deps/clojure/deps.json#L809, it was added in `1.11`.

Given that upgrading Clojure is generally safe, we should consider moving to `1.11.x` soon. `1.11.0` was released already almost 1 year ago https://clojure.org/news/2022/03/22/clojure-1-11-0.